src/App.spec.ts

```diff
@@ -1014,12 +1136,35 @@ describe('App', () => {
             // called
             assertionAggregator();
           });
-          dummyReceiverEvents.forEach(dummyEvent => fakeReceiver.emit('message', dummyEvent));
-          await delay();
+
+          await Promise.all(dummyReceiverEvents.map(event => fakeReceiver.sendEvent(event)));
 
           // Assert
           assert.equal(assertionAggregator.callCount, dummyReceiverEvents.length);
         });
+
+        it('should handle failures through the App\'s global error handler', async () => {
+          // Arrange
+          const fakePostMessage = sinon.fake.rejects(new Error('fake error'));
+          const overrides = buildOverrides([withPostMessage(fakePostMessage)]);
+          const App = await importApp(overrides); // tslint:disable-line:variable-name
+
+          const dummyMessage = { text: 'test' };
+          const dummyReceiverEvents = createChannelContextualReceiverEvents(dummyChannelId);
+
+          // Act
+          const app = new App({ receiver: fakeReceiver, authorize: sinon.fake.resolves(dummyAuthorizationResult) });
+          app.use(async (args) => {
+            // By definition, these events should all produce a say function, so we cast args.say into a SayFn
+            const say = (args as any).say as SayFn;
+            await say(dummyMessage);
```

Comment:
not quite sure why this test is calling `say()` (and setting up `fakePostMessage`). usually, we do this to allow us to assert that it was called with the arguments we expect later on in the test case. that isn't happening the way this is written now.

Reply:
This test was added by you actually 😛 https://github.com/slackapi/bolt/commit/d126b0b17b08d46138e36f1970f022f9fc9e5fbd#diff-a7a8d4758cfeb70999a809869b45fca0R524

I believe this used to fail from this issue: https://github.com/slackapi/bolt/issues/183

I had removed it in an earlier PR b/c I couldn't get it to pass without refactoring the error handling that this pr is doing

Reference: https://github.com/slackapi/bolt/pull/369#discussion_r370638637

SO, I can definitely refactor this, it's probably not necessary anymore actually. Wanted to give you a chance to dig into your memory to see if we should keep it.

Reply:
haha sorry for the confusion. my cruft is unrelated to the work in the PR so let's just resolve this and i can come back and clean things up later.